common/utils/src/main/scala/org/apache/spark/util/DayTimeIntervalUtils.java

```diff
@@ -0,0 +1,134 @@
+    if (field == DAY) {
+      return "DAY";
+    } else if (field == HOUR) {
+      return "HOUR";
+    } else if (field == MINUTE) {
+      return "MINUTE";
+    } else if (field == SECOND) {
+      return "SECOND";
+    } else {
+      throw new SparkException("Invalid field in day-time interval: " + field +
+              ". Supported fields are: DAY, HOUR, MINUTE, SECOND");
+    }
+  }
+
+  // Used to convert microseconds representing a day-time interval with given start and end fields
+  // to its ANSI SQL string representation. Throws a SparkException if startField or endField are
+  // out of bounds.
+  public static String toDayTimeIntervalANSIString(long micros, byte startField, byte endField)
+          throws SparkException {
+    String sign = "";
+    long rest = micros;
+    try {
+      String from = fieldToString(startField).toUpperCase();
+      String to = fieldToString(endField).toUpperCase();
+      String prefix = "INTERVAL '";
+      String postfix = startField == endField ? "' " + from : "' " + from + " TO " + to;
+      if (micros < 0) {
+        if (micros == Long.MIN_VALUE) {
+          // Especial handling of minimum `Long` value because negate op overflows `Long`.
+          // seconds = 106751991 * (24 * 60 * 60) + 4 * 60 * 60 + 54 = 9223372036854
+          // microseconds = -9223372036854000000L-775808 == Long.MinValue
+          String baseStr = "-106751991 04:00:54.775808000";
+          String firstStr = "-" + (startField == DAY ? Long.toString(MAX_DAY) :
+                  (startField == HOUR ? Long.toString(MAX_HOUR) :
+                          (startField == MINUTE ? Long.toString(MAX_MINUTE) :
+                                  Long.toString(MAX_SECOND) + ".775808")));
+          if (startField == endField) {
+            return prefix + firstStr + postfix;
+          } else {
+            int substrStart = startField == DAY ? 10 : (startField == HOUR ? 13 : 16);
+            int substrEnd = endField == HOUR ? 13 : (endField == MINUTE ? 16 : 26);
+            return prefix + firstStr + baseStr.substring(substrStart, substrEnd) + postfix;
+          }
+        } else {
+          sign = "-";
+          rest = -rest;
+        }
+      }
+      StringBuilder formatBuilder = new StringBuilder(sign);
+      ArrayList<Long> formatArgs = new ArrayList<>();
+      if (startField == DAY) {
+        formatBuilder.append(rest / MICROS_PER_DAY);
+        rest %= MICROS_PER_DAY;
+      } else if (startField == HOUR) {
+        formatBuilder.append("%02d");
+        formatArgs.add(rest / MICROS_PER_HOUR);
+        rest %= MICROS_PER_HOUR;
+      } else if (startField == MINUTE) {
+        formatBuilder.append("%02d");
+        formatArgs.add(rest / MICROS_PER_MINUTE);
+        rest %= MICROS_PER_MINUTE;
+      } else if (startField == SECOND) {
+        String leadZero = rest < 10 * MICROS_PER_SECOND ? "0" : "";
+        formatBuilder.append(leadZero + BigDecimal.valueOf(rest, 6)
```

Comment:
should use chained calls, like
```java
formatBuilder.append(leadZero).append(BigDecimal.valueOf(rest, 6).stripTrailingZeros().toPlainString());
```
otherwise `leadZero + BigDecimal.valueOf(rest, 6).stripTrailingZeros().toPlainString()` will result in another string concatenation."